tools/caffe.cpp

```diff
@@ -34,6 +34,13 @@ DEFINE_string(solver, "",
     "The solver definition protocol buffer text file.");
 DEFINE_string(model, "",
     "The model definition protocol buffer text file.");
+DEFINE_string(phase, "",
+    "Optional; network phase (TRAIN or TEST). Only used for 'time'.");
+DEFINE_int32(level, 0,
```

Comment:
Why are `level` and `stage` not honored for `caffe train` on the command line? I think it's reasonable for command line args to override the definition, just as the `-gpu` arg takes precedence over `solver_mode`.